vault/identity_store_entities.go

```diff
@@ -0,0 +1,463 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fatih/structs"
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/vault/helper/locksutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/mitchellh/mapstructure"
+)
+
+// entityPaths returns the API endpoints supported to operate on entities.
+// Following are the paths supported:
+// entity - To register a new entity
+// entity/id - To lookup, modify, delete and list entities based on ID
+// entity/merge/id - To merge entities based on ID
+func entityPaths(i *identityStore) []*framework.Path {
+	return []*framework.Path{
+		{
+			Pattern: "entity",
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeString,
+					Description: "Name of the entity",
+				},
+				"metadata": {
+					Type:        framework.TypeMap,
```

Comment:
If we do this, `abc=def==` would be a valid value given that the portion after the first `=` is base64 encoded. I guess its a trade off between CLI feature vs looks. I am leaning towards not worrying about how it looks and support this though CLI. Thanks for the suggestion.